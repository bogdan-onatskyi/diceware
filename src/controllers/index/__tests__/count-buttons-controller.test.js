import React from 'react';

import CountButtonsController from '../count-buttons-controller';
import passwordObject from '../../../models/password-object';

describe("CountButtonsController component", () => {
    it("should match saved snapshot", () => {
        passwordObject.wordArray.forEach((wordObject) => {
            wordObject.index = 0;
        });

        const wrapper = shallow(
            <CountButtonsController passwordObject={passwordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should have 1 prop", () => {
        const propsStr = 'passwordObject';
        const props = [];
        for (let prop in CountButtonsController.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});