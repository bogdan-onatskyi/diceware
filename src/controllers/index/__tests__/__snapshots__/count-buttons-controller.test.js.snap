// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountButtonsController component should match saved snapshot 1`] = `
<CountButtonsView
  handleResetAllWords={[Function]}
  handleUsedWords={[Function]}
  maxWords={8}
  usedWords={5}
/>
`;
